### DISCUSSION TITLE
Logged users permissions

### DISCUSSION THREAD

**Discussion by Walter-49** [prasathmani/tinyfilemanager#607] [Q&A]
Hello

I would like an authenticated user to be able to upload files and folders but not be able to delete them.
can you help me
thank you

**Comment by prasathmani** [comment]
create users as admin user, so user can all permission
https://github.com/prasathmani/tinyfilemanager/wiki/Security-and-User-Management

**Reply by Walter-49** [reply]
Hello
And how to remove the right to delete files or folders?

**Reply by prasathmani** [reply]
Currently the application does not have the option hide delete option.
but you can remove the delete button from code. just remove following highlighted lines
![image](https://user-images.githubusercontent.com/11345598/128446370-c160c9e1-c3d6-4964-a7b0-894450081f21.png)
![image](https://user-images.githubusercontent.com/11345598/128446446-194786fa-9dd7-43d7-b0a0-fa5dbb867b8a.png)

**Reply by Walter-49** [reply]
It will not be easier to add the upload file and folder function to the users because if I do as indicated even the administrator loses the delete function, right?

**Reply by prasathmani** [reply]
as tinyfilemanager as a single file, you can have one file for administrator and another one for user.

**Reply by Walter-49** [reply]
I would like to know how this kind of modification is implemented as my knowledge is very limited

**Reply by prasathmani** [reply]
just remove above mentioned lines from file

**Reply by Walter-49** [reply]
Ok i will try 
Thanks
